### ISSUE TITLE
Filters definition for YOLOv4 custom cfg

### ISSUE THREAD

**Issue by Avv22** [AlexeyAB/darknet#8740]
Hello,

In YOLOv4 [documentation](https://github.com/AlexeyAB/darknet), it's suggested to define the filters as follows:

`filters=(classes + 5)*<number of mask>`

![image](https://user-images.githubusercontent.com/45768058/219257649-c28357fa-bd02-4345-89bd-e7911b02fbd7.png)

But `anchors=9` in the cfg file [yolov4 cfg](https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov4-custom.cfg), so we why multiple by `num `and not by `anchor` please?

**Comment by stephanecharette** [comment]
Can you show us on what line in YOLOv4.cfg it says `anchors=9`?

Or do you mean there are 18 values in the v4 anchors, so 9 pairs?

Regardless, the important number you should be looking at is the number of masks, the line before the anchors.  This becomes much more obvious when working with configs that have more than 3 values in `mask=...` such as yolov4-p*.cfg.

So instead of `(classes + 5) * 3` the actual equation is `(classes + 5) * number_of_masks`.

If you use DarkMark, all of this is automatically fixed up for you in the .cfg file when you choose the option to generate the .cfg file.  You'll never have to manually edit a .cfg file again.  
![image](https://user-images.githubusercontent.com/5061352/219269433-9e211612-5aee-4d82-b350-db767f1d914a.png)

**Comment by Avv22** [comment]
> Can you show us on what line in YOLOv4.cfg it says `anchors=9`?
> 
> Or do you mean there are 18 values in the v4 anchors, so 9 pairs?
> 
> Regardless, the important number you should be looking at is the number of masks, the line before the anchors. This becomes much more obvious when working with configs that have more than 3 values in `mask=...` such as yolov4-p*.cfg.
> 
> So instead of `(classes + 5) * 3` the actual equation is `(classes + 5) * number_of_masks`.
> 
> If you use DarkMark, all of this is automatically fixed up for you in the .cfg file when you choose the option to generate the .cfg file. You'll never have to manually edit a .cfg file again. ![image](https://user-images.githubusercontent.com/5061352/219269433-9e211612-5aee-4d82-b350-db767f1d914a.png)

Yes. I mean by anchors the 18 values (9 anchor boxes). Why we don't use 9 instead of 3 masks when we calculate the filters please before the output layer?
